locals.tf

```diff
@@ -1,4 +1,5 @@
 locals {
-  env      = lower(terraform.workspace)
-  takeover = var.takeover == true && local.env == var.production_workspace ? true : false
+  env                    = coalesce(var.environment, lower(terraform.workspace))
+  production_environment = coalesce(var.production_environment, var.production_workspace)
```

Comment:
@paulschwarzenberger I've had to set `var.environment` and `var.production_environment` default to an empty string for now for coalesce to work. 

Should mean that it will always default to `terraform.workspace` and variable default `prd` if the new variables are not set.